### PR TITLE
adding docker registry docs for on-premise installation

### PR DESCRIPTION
**Pull request by noqcks**
This explains how to set up a docker registry with the on-prem installer. This is an optional step in the on-prem installation.

### CODE REVIEW

File: worker/on-premise/docker-registry.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+```
+[docker-registry]
+192.168.x.x
+```
+
+This registry will serve on port `6000`.
+
+# Usage
+
+You will need to alter your local docker before being able to push images to this registry. This is a little different depending on your OS -- please see [here](https://docs.docker.com/registry/insecure/) for how to set it up.
```
Comment (nikhilm) [inline worker/on-premise/docker-registry.md]: If our installer takes care of this, could we add a note saying so?

File: worker/on-premise/docker-registry.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+You can then docker push `hello` to your insecure registry
```
Comment (nikhilm) [inline worker/on-premise/docker-registry.md]: Nit: `:` or `.`

File: worker/on-premise/docker-registry.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+```
+docker push 192.168.x.x:6000/hello
+```
+
+Then register this image with Iron
```
Comment (nikhilm) [inline worker/on-premise/docker-registry.md]: Nit: `:` or `.`

File: worker/on-premise/docker-registry.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+```
+iron register -name test 192.168.x.x:6000/hello
+```
+
+You can then queue tasks to run like this
```
Comment (nikhilm) [inline worker/on-premise/docker-registry.md]: Nit: `:`

File: worker/on-premise/docker-registry.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+```
+iron worker queue test
+```
+
+For more information on the docker registry and how to create and push images, please see [this TL;DR](https://docs.docker.com/registry/#/tl-dr) from docker.
```
Comment (nikhilm) [inline worker/on-premise/docker-registry.md]: Docker Registry

File: worker/on-premise/docker-registry.md (hunk 1 above)
Comment (nikhilm) [inline worker/on-premise/docker-registry.md]: Docker

File: worker/on-premise/docker-registry.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+This will allow you to use a custom docker registry with an Iron On-Premise installation.
+
+There are two types of docker registries, insecure and secure. The one which you configure depends on your situation. However, due to limitations on controlling domains and certificates in an On Premise environment, we are only able to create an insecure registry.
+
+# Installation
+
+The only difference in workflow between an insecure registry and a secure registry is that you'll need to whitelist the registry from your local docker daemon.
+
+To set up a registry with the On Premise Installer, simply include a `[docker-registry]` host in your `./config/inventory.ini` with a proper server IP and the docker registry will be installed on that server.
```
Comment (nikhilm) [inline worker/on-premise/docker-registry.md]: Docker Registry

File: worker/on-premise/docker-registry.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+title: IronWorker On-Premise Docker Registry
+layout: default
+section: worker
+breadcrumbs:
+  - ['On-Premise', '/on-premise']
+---
+
+This will allow you to use a custom docker registry with an Iron On-Premise installation.
+
+There are two types of docker registries, insecure and secure. The one which you configure depends on your situation. However, due to limitations on controlling domains and certificates in an On Premise environment, we are only able to create an insecure registry.
```
Comment (nikhilm) [inline worker/on-premise/docker-registry.md]: Docker

Also, the "However, " is not necessary.

**Review by nikhilm — CHANGES_REQUESTED** [review]
Was a bit pedantic because this will eventually be user facing.

If you don't mind, could you wrap lines to 80 or 100 chars. Reads nicer when viewing it raw, and makes it easier to mark multiple review items separately if they are on separate lines but in the same sentence/paragraph. If you use vim, you can simply do this:

```
set textwidth=79
```

Then go to every paragraph and type `gq}` in normal mode.

Thanks.

File: worker/on-premise/docker-registry.md (hunk 1 above)
Comment (noqcks) [reply worker/on-premise/docker-registry.md]: Nah, our installer doesn't take care of this. In order to push/pull to an insecure registry one has to whitelist the Docker registry from their local Docker daemon on each machine they plan to push/pull with. It's not something that we can add into the installer.

**Comment by noqcks** [comment]
updated @nikhilm

**Comment by nikhilm** [comment]
So only the runners and possibly swapi images? In that case, it would be
nice to add that and also, if they need to restart Docker daemon after the
config change, just include

`systemctl restart docker` or something. Also similar commands to start the
runner (and possibly swapi) after they restart the daemon. Thanks.

**Comment by noqcks** [comment]
I mean that in order to push their custom images to the insecure registry they have to whitelist this registry from _their_ computer/server. We take care of runner/swapi whitelist during installation.

**Comment by nikhilm** [comment]
Ah! For dev machines and CI servers and things. Got it.

**Review by nikhilm — APPROVED** [review]

**Review by treeder — APPROVED** [review]

**Pull request merged**